src/Components/Blazor/WebAssembly.Authentication/src/Services/RemoteAuthenticationService.cs

```diff
@@ -120,8 +127,24 @@ public virtual async ValueTask<AccessTokenResult> GetAccessToken()
         /// <inheritdoc />
         public virtual async ValueTask<AccessTokenResult> GetAccessToken(AccessTokenRequestOptions options)
         {
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
             await EnsureAuthService();
-            return await _jsRuntime.InvokeAsync<AccessTokenResult>("AuthenticationService.getAccessToken", options);
+            var result = await _jsRuntime.InvokeAsync<AccessTokenResult>("AuthenticationService.getAccessToken", options);
+
+            var returnUrl = options.ReturnUrl != null ? _navigation.ToAbsoluteUri(options.ReturnUrl).ToString() : null;
+            var encodedReturnUrl = Uri.EscapeDataString(returnUrl ?? _navigation.Uri);
+            var redirectUrl = _navigation.ToAbsoluteUri($"{_options.AuthenticationPaths.LogInPath}?returnUrl={encodedReturnUrl}");
+
+            if (string.Equals(result.Status, AccessTokenResultStatus.RequiresRedirect, StringComparison.OrdinalIgnoreCase))
+            {
+                result.RedirectUrl = redirectUrl.ToString();
```

Comment:
Can we move L138-L140 inside this `if` block? Seems the don't need to be executed if we're not following this code path.

Secondly, why do we want `redirectUrl` to be absolute? Is there some reason why the default base-relative format won't work for us here?

Reply:
It makes it easier to reason about it in terms of security. You don't have to be thinking about where the url comes from and keeps consistency across the implementation.